File: 1222-remove-covered-intervals/remove-covered-intervals.java
class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> {
            if (a[0] != b[0]){
                return Integer.compare(a[0], b[0]);
            }
            return Integer.compare(b[1], a[1]);
        });
        int[] last = new int[2];
        last[0] = intervals[0][0];
        last[1] = intervals[0][1];
        int cnt = 1;
        for(int[] curr : intervals){
            if(!(last[0] <= curr[0] && last[1] >= curr[1])){
                last[0] = curr[0];
                last[1] = curr[1];
                cnt += 1;
            }
        }
        return cnt;
    }
}

File: 1222-remove-covered-intervals/remove-covered-intervals.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key = lambda x : (x[0], -x[1]))
        cnt = 1
        s = [intervals[0]]
        l = 1
        for i in range(1, len(intervals)):
            if not((s[-1][0] <= intervals[i][0]) and (s[-1][1] >= intervals[i][1])):
                s.append(intervals[i])
                l += 1
        return l